### PR TITLE
support ipex backend

### PR DESCRIPTION
**Pull request by violetch24**
# What does this PR do?
This PR is used to support INC IPEX features and add QA example.

**examples/neural_compressor/question-answering/run_qa.py**
- Add an argument "backend" in _OptimizationArguments_ to specify which backend would be used.
- Change _calib_dataloader_  from _get_train_dataloader()_ to _get_eval_dataloader()_ when using INC IPEX backend, in order to keep the form of _example_inputs_ identical.

**optimum/intel/neural_compressor/optimization.py**
- Load _pytorch_model.bin_ when using INC IPEX backend.

**optimum/intel/neural_compressor/quantization.py**
- Use _torch.jit.load_ to recover the int8 model quantized by INC IPEX backend, instead of _torch.load_.

**examples/neural_compressor/question-answering/README.md**
- Add INC IPEX backend example explanation

**Example**:
The following example applies INC IPEX backend:
python run_qa.py \
    --model_name_or_path bert-large-uncased-whole-word-masking-finetuned-squad \
    --dataset_name squad \
    --apply_quantization \
    --quantization_approach static \
    --do_train \
    --do_eval \
    --verify_loading \
    --output_dir /tmp/squad_output \
    --backend ipex

**Environment for this PR:**
Python 3.8.13
transformers 4.24.0
datasets 2.6.1
intel-extension-for-pytorch 1.12.300
torch 1.12.0

### CODE REVIEW

**Review by hshen14 — APPROVED** [review]
LGTM
